tests/helpers/perf_run_helper.rb

```diff
@@ -911,9 +913,53 @@ def get_perf_result
       puts
     end
 
+    save_average_compile_time("#{@archive_root}/avg_compile_time.txt", database)
+
     [perf, GatlingResult.new(gatling_assertions, mean_response_time)]
   end
 
+  def save_average_compile_time(file, puppetdb_host)
+    # Convert epoch values (which are created based on local time) to UTC time
+    # strings compatible with puppetdb queries.
+    start_time = Time.at(@start_epoch.to_i).getgm.strftime "%Y-%m-%d %H:%M:%S"
+    end_time = Time.at(@end_epoch.to_i).getgm.strftime "%Y-%m-%d %H:%M:%S"
+
+    query = <<~QUERY
+      query=["from", "reports",
+              ["extract", "metrics",
+                ["and",
+                  [">=", "start_time", "#{start_time}"],
+                  ["<=", "end_time", "#{end_time}"]]]]
+    QUERY
+    query = query.strip.gsub(/\n+/, " ")
+    query = query.gsub(/\s+/, " ")
+
+    curl = %W[
+      curl
+      -X GET
+      http://localhost:8080/pdb/query/v4
+      -d '#{query}'
+    ].join(" ")
+
+    result = nil
+    begin
+      bkr_res = on(puppetdb_host, curl)
+      data = JSON.parse(bkr_res.stdout.chomp)
+      data.delete_if { |report| report["metrics"]["data"].empty? }
+      config_retrieval_times = data.map do |report|
+        report["metrics"]["data"].select do |md|
+          md["category"] == "time" && (md["name"] == "config_retrieval" || md["name"] == "total")
+        end.first.fetch("value")
+      end
+      avg_config_retrieval_time = config_retrieval_times.reduce(0.0, :+) / config_retrieval_times.size
+      result = avg_config_retrieval_time.ceil
```

Comment:
We probably need more than an integer...  if the unit is seconds we would usually only get about 6...  So rounding could be significant.  what kind of numbers were you getting?

Reply:
The ceiling was pulled from Kishel's method https://github.com/tkishel/pe_tune/blob/f8270a784d2a6a96f1cd483ef95745dcf0a7e01e/lib/puppet_x/puppetlabs/tune/query.rb#L76-L102

I have no objection to using a float.

Reply:
yeah, Tom probably doesn't need as much precision as we probably will.  Though in the end AWS may end up making the extra precision pointless.  But we won't know until we start looking at it to see.

Reply:
I have updated this to use a float value.  The naming conventions have been changed.  In my research, I discovered that the report query times are not specifically catalog compile times.  The time was discovered to be in seconds, so this has been added to the payload written to disk as well.

Ref:
https://puppet.com/docs/puppet/latest/format_report.html
https://github.com/puppetlabs/puppet/blob/master/api/schemas/report.json#L254-L275
https://github.com/puppetlabs/puppet/blob/master/lib/puppet/transaction.rb#L50
https://github.com/puppetlabs/puppet/blob/master/lib/puppet/face/catalog.rb#L137-L139
https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util.rb#L519-L525